src/tiled/mapdocumentactionhandler.cpp

```diff
@@ -555,7 +606,19 @@ void MapDocumentActionHandler::updateActions()
     }
 
     mActionSelectAll->setEnabled(map);
-    mActionSelectNone->setEnabled(!selection.isEmpty());
+
+    if(currentLayer) {
+        if (currentLayer->asTileLayer()) {
+            mActionSelectNone->setEnabled(!selection.isEmpty());
+        } else if (currentLayer->asObjectGroup()) {
+            mActionSelectNone->setEnabled(mMapDocument->selectedObjects().count() > 0);
```

Comment:
Please also use `selectedObjectsCount` here, because `mMapDocument` may be null.